File: solutions/golang/snakeandladdergame/board.go
package snakeandladdergame

type Board struct {
	Size    int
	Snakes  []*Snake
	Ladders []*Ladder
}

func NewBoard() *Board {
	board := &Board{
		Size:    100,
		Snakes:  []*Snake{},
		Ladders: []*Ladder{},
	}
	board.initializeSnakesAndLadders()
	return board
}

func (b *Board) initializeSnakesAndLadders() {
	b.Snakes = append(b.Snakes, NewSnake(16, 6), NewSnake(48, 26), NewSnake(64, 60), NewSnake(93, 73))
	b.Ladders = append(b.Ladders, NewLadder(1, 38), NewLadder(4, 14), NewLadder(9, 31), NewLadder(21, 42),
		NewLadder(28, 84), NewLadder(51, 67), NewLadder(80, 99))
}

func (b *Board) GetNewPosition(position int) int {
	for _, snake := range b.Snakes {
		if snake.Start == position {
			return snake.End
		}
	}
	for _, ladder := range b.Ladders {
		if ladder.Start == position {
			return ladder.End
		}
	}
	return position
}

File: solutions/csharp/snakeandladdergame/Board.cs
using System;
using System.Collections.Generic;

namespace SnakeAndLadderGame
{
    public class Board
    {
        private const int BoardSize = 100;
        private readonly List<Snake> _snakes;
        private readonly List<Ladder> _ladders;

        public Board()
        {
            _snakes = new List<Snake>();
            _ladders = new List<Ladder>();
            InitializeSnakesAndLadders();
        }

        private void InitializeSnakesAndLadders()
        {
            // Initialize snakes
            _snakes.Add(new Snake(16, 6));
            _snakes.Add(new Snake(48, 26));
            _snakes.Add(new Snake(64, 60));
            _snakes.Add(new Snake(93, 73));

            // Initialize ladders
            _ladders.Add(new Ladder(1, 38));
            _ladders.Add(new Ladder(4, 14));
            _ladders.Add(new Ladder(9, 31));
            _ladders.Add(new Ladder(21, 42));
            _ladders.Add(new Ladder(28, 84));
            _ladders.Add(new Ladder(51, 67));
            _ladders.Add(new Ladder(80, 99));
        }

        public int GetBoardSize()
        {
            return BoardSize;
        }

        public int GetNewPositionAfterSnakeOrLadder(int position)
        {
            foreach (var snake in _snakes)
            {
                if (snake.Start == position)
                {
                    return snake.End;
                }
            }

            foreach (var ladder in _ladders)
            {
                if (ladder.Start == position)
                {
                    return ladder.End;
                }
            }

            return position;
        }
    }
}